src/libs/elektra/keyset.c

```diff
@@ -919,12 +926,26 @@ ssize_t ksAppendKey (KeySet * ks, Key * toAppend)
 		++ks->size;
 		if (ks->size >= ks->alloc)
 		{
-			if (ksResize (ks, ks->alloc * 2 - 1) == -1)
+			size_t newSize = ks->alloc * 2;
```

Comment:
```suggestion
			size_t newSize = ks->alloc == 0 ? KEYSET_SIZE : ks->alloc * 2;
			--newSize;
```

...and then remove the if below (not sure though why the `- 1` / `--` is needed)